modules/table/src/main/java/org/apache/ignite/internal/table/distributed/StorageUpdateHandler.java

```diff
@@ -100,6 +104,35 @@ public int partitionId() {
         return partitionId;
     }
 
+    /**
+     * Starts the handler doing necessary recovery.
+     */
+    public void start() {
+        recoverPendingRows();
+    }
+
+    private void recoverPendingRows() {
+        LOG.info("Recovering pending rows [tableId={}, partitionIndex={}]", storage.tableId(), storage.partitionId());
+
+        int count = 0;
+        try (Cursor<RowId> writeIntentRowIds = storage.getStorage().scanWriteIntents()) {
+            for (RowId rowId : writeIntentRowIds) {
+                ReadResult result = storage.getStorage().read(rowId, HybridTimestamp.MAX_VALUE);
+
+                if (!result.isEmpty() && result.isWriteIntent()) {
```

Comment:
Are we skipping tombstones? Why?

Please remove the `isEmpty` check, thank you!